src/main.cpp

```diff
@@ -198,6 +202,30 @@ namespace {
     set<int> setDirtyFileInfo;
 } // anon namespace
 
+bool HaveThinblockNodeConnections()
+{
+    BOOST_FOREACH(string& strAddNode, mapMultiArgs["-connect-thinblock"]) {
+        if(FindNode(strAddNode)) {
+            return true;
+        }
+    }
+    return false;
+
+}
+
+bool IsThinBlocksEnabled() 
+{
+    return GetBoolArg("-use-thinblocks", true);
+}
+
+bool IsChainNearlySyncd() 
+{
+    LOCK(cs_main);
+    if(chainActive.Height() < pindexBestHeader->nHeight - 2)
+        return false;
+    return true;
+}
+
```

Comment:
I think it's a great idea moving XThiin specific code to an external file, but I notice xthinblocks.* also contains code which is not XThin specific, or unlimited.* contains changes that are not unlimited specific and this will make it less likely to be merged into Core, due to the extra work someone will need to make to merge it...